Move UIImageView extension and its depending types to GravatarUI

pinarol opened the pull request:
Closes #149 

### Description

Move UIImageView extension and its depending types to GravatarUI

### Testing Steps

CI green

pinarol committed:
Move UIImageView to GravatarUI

pinarol committed:
More fixing

pinarol committed:
Remove unused images

pinarol committed:
Fix demo app

pinarol committed:
swiftformat

pinarol committed:
Add Bundle extension to the TestHelpers module

pinarol committed:
Remove TestHelpers

pinarol committed:
Remove unnecessary "public" usage

pinarol committed:
Update `GravatarImageSetMockResult.success as `.success`

etoledom commented on Demo/Demo/Gravatar-Demo/DemoUIImageViewExtensionViewController.swift:
```diff
@@ -8,6 +8,7 @@
 import Foundation
 import UIKit
 import Gravatar
+import GravatarUI
```
It would be great to only need to import `GravatarUI` and get everything from `Gravatar` also.
Like importing `UIKit` implicitly imports `Foundation`.

I guess we can add an @_exported import on `GravatarUI` if we want this to happen, though we can discuss it further without blocking this PR 👍

pinarol replied:
Good idea. I created a new issue to track this. https://github.com/Automattic/Gravatar-SDK-iOS/issues/154

etoledom reviewed (approved):
Looking great!

A couple of details:

When building the docs, there are two warnings of broken links:
![CleanShot 2024-04-02 at 22 08 28@2x](https://github.com/Automattic/Gravatar-SDK-iOS/assets/9772967/b084abde-455a-49eb-afe7-020e659e5cf6)

We need to restructure this whole description anyway, so I suggest only removing the links to get rid of the warnings, and we add a new ticket to update the docs and have a good doc base to continue the components work.

The other detail is on the code review. Not a blocker though 👍 

Feel free to merge 🎉

pinarol commented:
This is to track documentation update: https://github.com/Automattic/Gravatar-SDK-iOS/issues/155

pinarol committed:
Fix docs warnings

pinarol committed:
Merge branch 'trunk' into wppinar/move-uiimageview

pinarol commented:
I removed the links from GravatarWrapper. I removed the IdentifierProvider altogether because it doesn't exist in our code base. It must be a leftover from a PR.
